### PR TITLE
Fix for lead being overwritten

### PR DESCRIPTION
**Pull request by nahidalgo**
https://tree.taiga.io/project/nahidalgo-wopta/issue/246

### CODE REVIEW

File: broker/lead.go [hunk 1]
```diff
@@ -115,6 +109,41 @@ func lead(authToken models.AuthToken, policy *models.Policy) error {
 	return err
 }
 
+func checkIfPolicyIsLead(policy *models.Policy) error {
+	var recoveredPolicy models.Policy
+	log.Printf("[lead] found lead for existing policy %s", policy.Uid)
```
Comment (diogo-wopta) [inline broker/lead.go]: move log to line 129, after we successfully retrieved the policy from the DB

File: broker/lead.go [hunk 2]
```diff
@@ -115,6 +109,41 @@ func lead(authToken models.AuthToken, policy *models.Policy) error {
...
+		return nil
+	}
+
+	if recoveredPolicy.Status != models.PolicyStatusPartnershipLead && recoveredPolicy.Status != models.PolicyStatusInitLead {
+		log.Printf("[lead] error policy %s is not a lead", policy.Uid)
+		return errors.New("policy is not a lead")
+	}
+
+	policy.CreationDate = recoveredPolicy.CreationDate
+	policy.Updated = recoveredPolicy.Updated
```
Comment (diogo-wopta) [inline broker/lead.go]: updated field can be removed as it will be set to now if the changes are successful

**Review by diogo-wopta — APPROVED** [review]

File: broker/lead.go (hunk 1 above)
Comment (nahidalgo) [reply broker/lead.go]: Done in https://github.com/wopta/goworkspace/pull/121/commits/0d1cfbfbdb3a1bfc445c793fda5ffeef4f83cc4c

File: broker/lead.go (hunk 2 above)
Comment (nahidalgo) [reply broker/lead.go]: Done in https://github.com/wopta/goworkspace/pull/121/commits/0d1cfbfbdb3a1bfc445c793fda5ffeef4f83cc4c

**Review by diogo-wopta — APPROVED** [review]

**Pull request merged**